core/src/player.rs

```diff
@@ -1252,15 +1252,59 @@ impl Player {
                 ));
             }
 
+            let mut new_over_object_updated = false;
             // Cancel hover if an object is removed from the stage.
             if let Some(hovered) = context.mouse_over_object {
                 if !context.is_action_script_3() && hovered.as_displayobject().avm1_removed() {
                     context.mouse_over_object = None;
+                    if let Some(new_object) = new_over_object {
+                        if new_object.as_displayobject().id() == hovered.as_displayobject().id()
```

Comment:
if these conditions "id, depth and movie match" are duplicated, we might wanna reuse these as some function that checks for "kinda-sorta-equivalence" of displayobjects

Reply:
Added a function to checck for the same depth and id, but not to check for the same movie because that is only be done in one place.